CompanyEmployees.Application/Validators/CheckInAttendanceCommandValidator.cs

```diff
@@ -0,0 +1,37 @@
+using CompanyEmployees.Application.Commands;
+using FluentValidation;
+using FluentValidation.Results;
+
+namespace CompanyEmployees.Application.Validators;
+
+public sealed class CheckInAttendanceCommandValidator : AbstractValidator<CheckInAttendanceCommand>
+{
+    public CheckInAttendanceCommandValidator()
+    {
+        RuleFor(a => a.Attendance.EmployeeId)
+            .NotEmpty()
+            .WithMessage("Employee ID is required.");
+
+        RuleFor(a => a.Attendance.CheckInTime)
+            .NotEmpty()
+            .WithMessage("Check-in time is required.");
+
+        RuleFor(a => a.Attendance.Status)
+            .NotEmpty()
+            .MaximumLength(10)
+            .WithMessage("Status is required and must not exceed 10 characters.");
+
+        RuleFor(a => a.Attendance.Notes)
```

Comment:
The `When` condition here is redundant because `MaximumLength` validation already handles null and empty values gracefully. Including the `When` clause unnecessarily complicates the rule.

```suggestion
RuleFor(a => a.Attendance.Notes)
    .MaximumLength(200)
    .WithMessage("Notes must not exceed 200 characters.");
```